### PR TITLE
Port E2E test automation from release-0.4 branch

### PR DESCRIPTION
**Pull request by randomvariable**
Ported E2E test automation added in release-0.4 branch to master branch.
This automation includes conformance tests also.

Verified the test results in local execution and all tests succeeded.

Authored-by: T V Kutumba Rao. Sidrala <sirao@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Finishes up #1501 

/hold 
Cannot merge before https://github.com/kubernetes-sigs/cluster-api/issues/2396

### CODE REVIEW

**Comment by randomvariable** [comment]
Prow probably has a new enough version of kind to default the image, so will remove hold if the test progresses far enough

File: config/manager/manager_image_patch.yaml [hunk 1]
```diff
@@ -8,5 +8,5 @@ spec:
     spec:
       containers:
       # Change the value of image field below to your controller image URL
-      - image: gcr.io/k8s-staging-cluster-api-aws/cluster-api-aws-controller:latest
+      - image: gcr.io/stone-poetry-490/cluster-api-aws-controller-amd64:dev
```
Comment (detiber) [inline config/manager/manager_image_patch.yaml]: Added by mistake?

File: config/manager/manager_pull_policy.yaml [hunk 2]
```diff
@@ -8,4 +8,4 @@ spec:
     spec:
       containers:
       - name: manager
-        imagePullPolicy: Always
+        imagePullPolicy: IfNotPresent
```
Comment (detiber) [inline config/manager/manager_pull_policy.yaml]: Added by mistake?

File: config/manager/manager_image_patch.yaml (hunk 1 above)
Comment (randomvariable) [reply config/manager/manager_image_patch.yaml]: argh

File: config/manager/manager_pull_policy.yaml (hunk 2 above)
Comment (randomvariable) [reply config/manager/manager_pull_policy.yaml]: need to sort out that `make test-e2e` so it doesn't do this.

Comment (randomvariable) [reply config/manager/manager_pull_policy.yaml]: excised

**Comment by detiber** [comment]
/approve
/lgtm
/hold
holding pending the e2e prow job

**Comment by randomvariable** [comment]
Did this time out during the "Creating cluster after reaching vpc maximum limit" or something else?

**Comment by detiber** [comment]
It looks like it made it through 13 of the 16 Specs prior to timing out. It was indeed the "Creating cluster after reaching vpc maximum limit" spec that was in progress when it hit the 4hr limit

**Comment by randomvariable** [comment]
Yeah, had a look through the CAPI logs, looks like it was in the process of mopping up the cluster for that test when it got killed

**Comment by randomvariable** [comment]
Can take a look at forward porting https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1573 next week?

**Comment by randomvariable** [comment]
Baton passed to #1596 

/close

**Pull request closed without merging**